### ISSUE TITLE
Plugin missing on session change

### ISSUE THREAD

**Issue by DIOHz0r** [glpi-project/glpi#3688]
* Steps to reproduce (which actions have you made) :

1. Install and activate any plugin
2. Make the API where a user has a different session
3. Check the plugins available on session 

* Expected result :

The plugins available on the first login must be available on session change

* Actual result :

The plugins are lost

* Screenshot of the problem (if pertinent) :

Before session check....
![imagen](https://user-images.githubusercontent.com/1426313/37119047-967569ac-222c-11e8-9acc-ce2b3d3aedd4.png)

After session change...
![imagen](https://user-images.githubusercontent.com/1426313/37119094-c4b6d044-222c-11e8-9d2e-1019310c2907.png)

* Your GLPI setup (you can find it in Setup > General menu, System tab) :
GLPi 9.2.2

**Comment by fgendorf** [comment]
Same problem here

**Comment by edgardmessias** [comment]
For REST: Add in `apirest.php`, before `include ('./inc/includes.php');`:

```php
if(isset($_GET['session_token'])){
    session_id($_GET['session_token']);
}
```

**Comment by edgardmessias** [comment]
Solution for REST and XMLRPC: #3692

**Comment by DIOHz0r** [comment]
Hello @edgardmessias ,

I did another PR (#3697) with a simple a better fix for this issue IMO with. Let's wait until @trasher review both PR and check which one is better.

**Comment by edgardmessias** [comment]
My solution initialize only after detect the session id.
Your solution, the plugins are initialize twice times and on each request, a new session is created and destroyed.

**Comment by DIOHz0r** [comment]
@edgardmessias, Why do you say that the plugins initialize twice and on each request?, If the plugin array exist on session they won't be initialized again, also I'm leaving all the logic about security like cross, sanitize parameters, check if the API is enable, etc. that you removed

**Comment by edgardmessias** [comment]
@DIOHz0r, to you understand, the real problem is the double session.
On API, a temporary session is created, call all `plugin_init_<plugin>`, destroy and load the session from token.

In `plugin_init_<plugin>`, user is not logged, and if some plugin use `$_SESSION`, the user is none

**Comment by edgardmessias** [comment]
@DIOHz0r, about security check, see https://github.com/edgardmessias/glpi/blob/api_session/inc/api.class.php#L111

Note: Current code return a HTML with message `api disable`, my solution return JSON for apirest and XML for apixmlrpc

**Comment by edgardmessias** [comment]
The #3692 close this issue
